dowhy/gcm/validation.py

```diff
@@ -0,0 +1,157 @@
+                      'edge_dependence_test': {'Y': {'p_value': 0.5, 'fdr_adjusted_p_value': 0.5, 'success': True}}}}]
+    """
+    is_dag_valid = True
+    validation_summary = dict()
+    all_p_values = []
+
+    for node in causal_graph.nodes:
+        parents = get_ordered_predecessors(causal_graph, node)
+        non_descendants = _get_non_descendants(causal_graph, node, exclude_parents=True)
+
+        lmc_test_result = dict()
+        if parents and non_descendants:
+            # test local Markov condition, null=conditional independence
+            lmc_p_value = conditional_independence_test(data[node].values,
+                                                        data[non_descendants].values,
+                                                        data[parents].values)
+            lmc_test_result = dict(p_value=lmc_p_value)
+            all_p_values.append(lmc_p_value)
+
+        # test edge dependence, null=independence
+        edge_dependence_result = dict()
+        for parent in parents:
+            edge_p_value = independence_test(data[parent].values, data[node].values)
+            edge_dependence_result[parent] = dict(p_value=edge_p_value)
+            all_p_values.append(edge_p_value)
+
+        validation_summary[node] = dict(
+            local_markov_test=lmc_test_result,
+            edge_dependence_test=edge_dependence_result
+        )
+
+    if fdr_control_method is None:
+        successes = np.array(all_p_values) <= significance_level
+        adjusted_p_values = np.empty(len(successes))
+        adjusted_p_values[:] = np.nan
+    else:
+        multipletests_result = multipletests(all_p_values, significance_level, method=fdr_control_method)
+        successes = multipletests_result[0]
+        adjusted_p_values = multipletests_result[1]
+
+    # The order of the p-values added to the list is deterministic.
+    index = 0
+    for node in causal_graph.nodes:
+        if 'p_value' in validation_summary[node]['local_markov_test']:
+            validation_summary[node]['local_markov_test']['fdr_adjusted_p_value'] \
+                = adjusted_p_values[index]
+            validation_summary[node]['local_markov_test']['success'] \
+                = not successes[index]
+            is_dag_valid &= not successes[index]
+            index += 1
+
+        for parent in get_ordered_predecessors(causal_graph, node):
+            validation_summary[node]['edge_dependence_test'][parent]['fdr_adjusted_p_value'] \
+                = adjusted_p_values[index]
+            validation_summary[node]['edge_dependence_test'][parent]['success'] \
+                = successes[index]
+            is_dag_valid &= successes[index]
+            index += 1
+    if is_dag_valid:
+        return RejectionResult.NOT_REJECTED, validation_summary
+    else:
+        return RejectionResult.REJECTED, validation_summary
+
+
+def validate_pnl_assumptions(causal_model: InvertibleStructuralCausalModel,
```

Comment:
a more accessible name would be `validate_post_nonlinear_model`, or to be consistent with refutation, `refute_post_nonlinear_model`. I can imagine two high level API functions: refute_model and refute_graph and then functions like this are specific implementations of them.

Reply:
I renamed it to ```refute_invertible_model```, since this is what it actually does and it should be more consistent then with the rest. In this regard, I also revised the docstring.